rpc/src/jsonrpc.rs

```diff
@@ -30,3 +31,17 @@ pub struct Pool {
     /// the used capacity in bytes
     pub used: u64,
 }
+use std::str::FromStr;
+impl FromStr for ShareProtocol {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<ShareProtocol, ()> {
+        match s.to_lowercase().trim() {
+            "none" => Ok(ShareProtocol::None),
```

Comment:
perhaps replace the arms to handle multiple cases like   "nvnf" | "NVMF" | "Nvmf" and so on.
NVMF because that is how it is defined in rpc/proto/mayastor.proto
Nvmf because that is how it is converted to rust see target/debug/build/rpc-*/out/mayastor.rs and mayastor/src/bdev/nexus/nexus_share.rs (in tjoshum).
mctl will primarily be used by developers so allowing them to use symbols they are familiar with makes sense.